modules/analysis/src/model.rs

```diff
@@ -0,0 +1,193 @@
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "graph_count {}", self.graph_count)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct DepNode {
+    pub purl: String,
+    pub name: String,
+    pub deps: Vec<DepNode>,
+}
+impl fmt::Display for crate::model::DepNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.name)
+    }
+}
+impl DepNode {
+    pub fn new(purl: String, name: String, contains: Vec<DepNode>) -> DepNode {
+        DepNode {
+            purl,
+            name,
+            deps: contains,
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ToSchema, serde::Serialize)]
+pub struct PackageNode {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+}
+impl fmt::Display for PackageNode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.name)
+    }
+}
+
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct AdvisoryGraphSummary {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+
+    pub ancestors: Vec<PackageNode>,
+}
+impl AdvisoryGraphSummary {
+    pub fn new(
+        purl: String,
+        name: String,
+        published: String,
+        document_id: String,
+        product_name: String,
+        product_version: String,
+        ancestors: Vec<PackageNode>,
+    ) -> AdvisoryGraphSummary {
+        AdvisoryGraphSummary {
+            purl,
+            name,
+            published,
+            document_id,
+            product_name,
+            product_version,
+            ancestors,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, ToSchema)]
+pub struct AdvisoryDepSummary {
+    pub purl: String,
+    pub name: String,
+    pub published: String,
+    pub document_id: String,
+    pub product_name: String,
+    pub product_version: String,
+    pub deps: Option<Vec<DepNode>>,
+}
+impl crate::model::AdvisoryDepSummary {
+    pub fn new(
+        purl: String,
+        name: String,
+        published: String,
+        document_id: String,
+        product_name: String,
+        product_version: String,
+        contains: Vec<DepNode>,
+    ) -> crate::model::AdvisoryDepSummary {
+        crate::model::AdvisoryDepSummary {
+            purl,
+            name,
+            published,
+            document_id,
+            product_name,
+            product_version,
+            deps: Some(contains),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct GraphMap {
+    map: HashMap<String, Graph<PackageNode, Relationship, petgraph::Directed>>,
+}
+
+// Static variables for the singleton instance
+static mut INSTANCE: Option<Arc<RwLock<GraphMap>>> = None;
```

Comment:
If we trade it around to Arc<RwLock<Option<GraphMap>> is that easier to avoid the static mut and it's related unsafe?

Reply:
oh that would prob be better ... I did it this way first and it worked and moved on ;) thx for the prod.

Reply:
I opted to use lazy_static